ocaml/libs/tracing/tracing.ml

```diff
@@ -15,13 +15,65 @@ module D = Debug.Make (struct let name = "tracing" end)
 
 open D
 
+module W3CBaggage = struct
+  module Key = struct
+    let is_valid_key str =
+      let is_tchar = function
+        | '0' .. '9'
+        | 'a' .. 'z'
+        | 'A' .. 'Z'
+        | '!'
+        | '#'
+        | '$'
+        | '%'
+        | '&'
+        | '\''
+        | '*'
+        | '+'
+        | '-'
+        | '.'
+        | '^'
+        | '_'
+        | '`'
+        | '|'
+        | '~' ->
+            true
+        | _ ->
+            false
+      in
+      String.to_seq str |> Seq.for_all (fun c -> is_tchar c)
```

Comment:
Why not `String.for_all`? This should be most efficient because it does not allocate.

Reply:
same question, what's the benefit of using sequence here?